IJ gradle plugin fails with JAR dependency from Ultimate not found error

Very simple configuration to reproduce is here:
https://github.com/ivanporty/intellij-gradle-plugin-test

Just run:
`./gradlew compileJava`

Fails with:
FAILURE: Build failed with an exception.

* What went wrong:
Could not resolve all files for configuration ':compileClasspath'.
> Could not find lib/duplicates-groovy.jar (org.jetbrains.plugins:Groovy:2018.1-withSources).
  Searched in the following locations:
      .gradle/caches/modules-2/files-2.1/com.jetbrains.intellij.idea/ideaIC/2018.1/d082978a55b398ef544e7daee1dd6228b88dbfc1/ideaIC-2018.1/plugins/Groovy/lib/duplicates-groovy.jar
      .gradle/caches/modules-2/files-2.1/com.jetbrains.intellij.idea/plugins.jetbrains.com/Groovy-2018.1-withSources/lib/duplicates-groovy.jar
      .gradle/caches/modules-2/files-2.1/com.jetbrains.intellij.idea/ideaIC/2018.1/41575d90c6dfac0e36e5c9328b89ad85de1da2c1/ideaIC-2018.1-sources.jar/lib/duplicates-groovy-2018.1-withSources.jar

--- a/src/main/kotlin/org/jetbrains/intellij/jbr/JbrResolver.kt
+++ b/src/main/kotlin/org/jetbrains/intellij/jbr/JbrResolver.kt
@@ -259,6 +259,7 @@ abstract class JbrResolver @Inject constructor(
                 val buildNumber = Version.parse(buildNumberString)
                 val isJava8 = majorVersion.startsWith("8")
                 val isJava17 = majorVersion.startsWith("17")
+                val isJava21 = majorVersion.startsWith("21")
 
                 val oldFormat = prefix == "jbrex" || isJava8 && buildNumber < Version.parse("1483.24")
                 if (oldFormat) {
@@ -271,7 +272,7 @@ abstract class JbrResolver @Inject constructor(
                 val arch = jbrArch ?: arch(isJava8)
                 if (prefix.isEmpty()) {
                     prefix = when {
-                        isJava17 -> "jbr_jcef-"
+                        isJava17 || isJava21 -> "jbr_jcef-"
                         isJava8 -> "jbrx-"
                         operatingSystem.isMacOsX && arch == "aarch64" -> "jbr_jcef-"
                         buildNumber < Version.parse("1319.6") -> "jbr-"

--- a/src/test/kotlin/org/jetbrains/intellij/jbr/JbrResolverTest.kt
+++ b/src/test/kotlin/org/jetbrains/intellij/jbr/JbrResolverTest.kt
@@ -18,6 +18,15 @@ class JbrResolverTest : IntelliJPluginSpecBase() {
     fun `resolve 11_0_11b1536_2`() = testJbrResolving("11_0_11b1536.2", "jbr_jcef-11_0_11-$platform-$arch-b1536.2")
 
     @Test
+    fun `resolve 17_0_10b1087_17`() = testJbrResolving("17.0.10b1087.17", "jbr_jcef-17.0.10-$platform-$arch-b1087.17")
+
+    @Test
+    fun `resolve 21_0_2b346_3`() = testJbrResolving("21.0.2b346.3", "jbr_jcef-21.0.2-$platform-$arch-b346.3")
+
+    @Test
+    fun `resolve 21_0_2b341_4`() = testJbrResolving("21.0.2b341.4", "jbr_jcef-21.0.2-$platform-$arch-b341.4")
+
+    @Test
     fun `resolve jbrsdk-11_0_13b1751_16`() = testJbrResolving("jbrsdk-11_0_13b1751.16", "jbrsdk-11_0_13-$platform-$arch-b1751.16")
 
     @Test

